tic_tac_toe/game.py

```diff
@@ -87,18 +136,57 @@ def move(game, player, position):
     checks before the actual movement is done.
     After registering the movement it must check if the game is over.
     """
-    pass
+    board = game['board']
+    if game['winner'] or _board_is_full(board):
+        raise InvalidMovement("Game is over.")
+        
+    elif not _position_is_valid(position):
+        raise InvalidMovement('Position out of range.')
+    
+    elif not _position_is_empty_in_board(position, board):
+        raise InvalidMovement('Position already taken.')
+    
+    elif get_next_turn(game) != player:
+        raise InvalidMovement('"{}" moves next'.format(game['next_turn']))
+    
+    if player == game['player1']:
+        game['next_turn'] = game['player2']
+    else:
+        game['next_turn'] = game['player1']
+    
+    board[position[0]][position[1]] = player
+    
+    winner = _check_winning_combinations(board, player)
+    
+    if winner:
+        game['winner'] = winner
+        game['next_turn'] = None
+        raise GameOver('"{}" wins!'.format(winner))
+    
+    elif _board_is_full(board):
+        game['next_turn'] = None
+        raise GameOver('Game is tied!')
 
 
 def get_board_as_string(game):
     """
     Returns a string representation of the game board in the current state.
     """
-    pass
+    board_string = """
+{}  |  {}  |  {}
+--------------
+{}  |  {}  |  {}
+--------------
+{}  |  {}  |  {}
+"""
+        
+    board = game['board']
+    
+    return board_string.format(board[0][0],board[0][1],board[0][2],board[1][0],board[1][1],board[1][2],board[2][0],board[2][1],board[2][2])
```

Comment:
You could simplify this using `chain` from the `itertools` builtin module:
```
from itertools import chain
return board_string.format(*chain.from_iterable(game['board'])
```
This flattens a multi-dimensional list one level (two-dimension to one dimension) and then expands it to fill in your format parameters.